.github/workflows/dm.yml

```diff
@@ -0,0 +1,184 @@
+
+    runs-on: ubuntu-latest
+    name: install-with-conda
+
+    timeout-minutes: 5
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Conda
+      uses: s-weigand/setup-conda@v1
+      with:
+        update-conda: true
+        python-version: 3.7
+    - run: conda --version
+    - run: python --version
+    - run: which python
+
+    - name: Create conda environment
+      run: |
+        conda env create -f environment.yml -n test-environment
+        source activate test-environment
+        python -m pip install --no-deps -e .
+
+    - name: Test package import
+      run: |
+        source activate test-environment
+        python -c "import aiida"
+
+  tests:
+
+    needs: [install-with-pip, install-with-conda]
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.5, 3.6, 3.7, 3.8]
+        backend: ['django', 'sqlalchemy']
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: CasperWA/postgresql-action@v1.2
+      with:
+        postgresql version: '10'
+        postgresql db: test_${{ matrix.backend }}
+        postgresql user: postgres
+        postgresql password: ''
+        postgresql auth: trust
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install system dependencies
+      run: |
+        wget -O - "https://github.com/rabbitmq/signing-keys/releases/download/2.0/rabbitmq-release-signing-key.asc" | sudo apt-key add -
+        echo 'deb https://dl.bintray.com/rabbitmq-erlang/debian bionic erlang' | sudo tee -a /etc/apt/sources.list.d/bintray.rabbitmq.list
+        echo 'deb https://dl.bintray.com/rabbitmq/debian bionic main' | sudo tee -a /etc/apt/sources.list.d/bintray.rabbitmq.list
+        sudo rm -f /etc/apt/sources.list.d/dotnetdev.list /etc/apt/sources.list.d/microsoft-prod.list
+        sudo apt update
+        sudo apt install postgresql-10 rabbitmq-server graphviz
+        sudo systemctl status rabbitmq-server.service
+
+    - name: Install python dependencies
+      run: |
+        pip install --upgrade pip
+        pip install numpy==1.17.4
+        pip install -e .[atomic_tools,docs,notebook,rest,testing]
+        reentry scan
+
+    - name: Setup environment
+      env:
+        AIIDA_TEST_BACKEND: ${{ matrix.backend }}
+      run:
+        .github/workflows/setup.sh
+
+    - name: Run test suite
+      env:
+        AIIDA_TEST_BACKEND: ${{ matrix.backend }}
+      run:
+        .github/workflows/tests.sh
+
+    - name: Freeze test environment
+      run: |
+        pip freeze | tee requirements-${{ matrix.backend }}-py-${{ matrix.python-version }}.txt
+
+    - name: Upload requirements.txt
+      uses: actions/upload-artifact@v1
+      with:
+        name: requirements.txt
+        path: requirements-${{ matrix.backend }}-py-${{ matrix.python-version }}.txt
+
+  update-requirements:
```

Comment:
One thing is to check daily that aiida installs fine & tests run - that's fine.
Updating the fixed requirements.txt I see as less urgent and, as mentioned before, I feel it is almost better not to run it too often. Last time I checked, overall aiida-core had more than a hundred dependencies, i.e. I suspect this job will be triggered very often.

Since the main purpose of the requirements.txt (at least to me) is reproducibility, I would suggest to tie it to a release - so, it could e.g. be a separate workflow that only runs on PRs to `master` (and perhaps even allow failures in order not to block releases if, in between reviews, a dependency is updated).

Reply:
I think that's a good call. I'll update the workflow accordingly.

Reply:
Done.